app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def decrease_health(self, amount: int) -> None:
+        self.health -= amount
+        if self.health <= 0:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+        if self.health <= 0:
+            del self
```

Comment:
The line `del self` is not a proper way to handle object deletion in Python. This does not remove the object from the `Animal.alive` list, and it may lead to unexpected behavior. Consider using a method to properly manage the object's lifecycle, such as removing it from the list if necessary.